spec/models/vm_reconfigure_task_spec.rb

```diff
@@ -19,48 +19,86 @@
-      end
+  describe ".base_model" do
+    it "should return VmReconfigureTask" do
+      expect(VmReconfigureTask.base_model).to eq(VmReconfigureTask)
     end
+  end
 
-    describe "#after_request_task_create" do
-      it "should set the task description" do
-        task.after_request_task_create
-        expect(task.description).to include("VM Reconfigure for: #{vm} - ")
-      end
+  describe "#after_request_task_create" do
+    it "should set the task description" do
+      task.after_request_task_create
+      expect(task.description).to include("VM Reconfigure for: #{vm} - ")
     end
   end
 
+  context "Single Disk add " do
+    let(:request_options) { {:disk_add => [{"disk_size_in_mb" => "33", "persistent" => "true"}]} }
+    let(:description_partial) { "Add Disks: 1 : #{request.options[:disk_add][0]["disk_size_in_mb"].to_i.megabytes.to_s(:human_size)} " }
+
+    it_behaves_like ".get_description"
+  end
+
   context "Multiple Disk add " do
-    let(:request) do
-      VmReconfigureRequest.create(:requester    => user,
-                                  :options      => {:src_ids  => [vm.id],
-                                                    :disk_add => [{"disk_size_in_mb" => "33", "persistent" => "true"},
-                                                                  {"disk_size_in_mb" => "44", "persistent" => "true"}]},
-                                  :request_type => 'vm_reconfigure')
+    let(:request_options) do
+      {:disk_add => [{"disk_size_in_mb" => "33", "persistent" => "true"},
+                     {"disk_size_in_mb" => "44", "persistent" => "true"}]}
+    end
+    let(:description_partial) do
+      "Add Disks: 2 : #{request.options[:disk_add][0]["disk_size_in_mb"].to_i.megabytes.to_s(:human_size)}, "\
+      "#{request.options[:disk_add][1]["disk_size_in_mb"].to_i.megabytes.to_s(:human_size)} "
+    end
+
+    it_behaves_like ".get_description"
+  end
+
+  context "Hardware Properties" do
+    let(:request_options) do
+      {:vm_memory         => 512,
+       :number_of_sockets => 2,
+       :cores_per_socket  => 4,
+       :number_of_cpus    => 1}
     end
+    let(:description_partial) { "Memory: 512 MB, Processor Sockets: 2, Processor Cores Per Socket: 4, Total Processors: 1" }
+
+    it_behaves_like ".get_description"
+  end
 
-    describe "#self.get_description" do
-      it "should get the task description" do
-        expect(VmReconfigureTask.get_description(request)).to eq("VM Reconfigure for: #{vm} - Add Disks: 2 : #{request.options[:disk_add][0]["disk_size_in_mb"].to_i.megabytes.to_s(:human_size)}, "\
-          "#{request.options[:disk_add][1]["disk_size_in_mb"].to_i.megabytes.to_s(:human_size)} ")
-      end
+  context "Disk remove/resize" do
+    let(:request_options) do
+      {:disk_remove => [1],
+       :disk_resize => [1, 2]}
```

Comment:
@lfu Do you have useful examples of what the data would look like for these properties?  I just added a simple array knowing that the `length` method is called against these properties when building the string.  See: https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_reconfigure_task.rb#L34

Same question applies to "Network" and "CDROM" below.

Reply:
Unfortunately I don't. But it seems in the format like this:
```
:disk_resize => [{:disk_name => filename, :disk_size_in_mb => disk_size}],
:network_adapter_add    => [
          { :cloud_network => 'vApp Network Name (vapp name)', :name => 'VM Name#NIC#2' },
          { :cloud_network => nil, :name => 'VM Name#NIC#3' }
],
:network_adapter_remove => [{ :network => { :name => 'VM Name#NIC#0' } }],
:cdrom_connect => [
      { :device_name => "CD/DVD drive 1",
        :filename    => "[NFS Share] ISO/centos.iso",
        :storage_id  => storage.id,
      }
]
```